### PR TITLE
Use gulp-macdmg for Big Sur compatibility

### PR DESCRIPTION
**Pull request by israelshirk**
Resolves build issues related to...

https://github.com/LinusU/fs-xattr/issues/27

### CODE REVIEW

**Comment by mikeller** [comment]
Not sure why switching to a different package that has 50 times less use and was last released 7 months ago should be fixing an issue with Big Sur which was released weeks ago - the issue you are referencing in 2019, so it does not seem to be having relevance to Big Sur either.

Can you provide more information as to what this change does, and what issue it is fixing?

**Comment by israelshirk** [comment]
At a high level, this is the minimum amount of invasiveness I could find with the time I had to get blheli-configurator working on current OS X (Big Sur) with current installs of Brew, Node, etc.

Appdmg is no longer maintained, so while it may have a larger popularity, it no longer supports default installs of OS X since Big Sur RC 1...  Macdmg is a fork of Appdmg which is currently maintained and keeps backwards-compatibility to the Appdmg APIs such that there’s a minimum amount of change necessary to keep apps up to date.

At a lower level it is technically fixing a compatibility issue with fs-xattr, which is a requirement for macdmg/appdmg/etc on filesystems with extended attributes (HFS/APFS).  fs-xattr v0.17 is pinned by appdmg and is no longer compatible with current OS X/node/xcode, while fs-xattr v0.30 pinned by macdmg.  There are other fixes/improvements as well, which I was less interested in but seemed benign enough

So, long story short, it builds.

(Please pardon if appdmg/macdmg are swapped in this comment, it’s been a heck of a day)

**Comment by mikeller** [comment]
@israelshirk: Makes sense. But it looks like even with this change building does not work under Big Sur:

```
mikeller@iMac blheli-configurator % yarn gulp release   
yarn run v1.22.5
$ gulp release
[04:47:00] Using gulpfile ~/git/blheli-configurator/gulpfile.js
[04:47:00] Starting 'release'...
[04:47:00] Starting 'clean_release'...
[04:47:00] Starting 'clean_apps'...
[04:47:00] Starting 'clean_dist'...
[04:47:00] Starting 'build_js'...
[04:47:00] Finished 'clean_release' after 105 ms
[04:47:04] Finished 'build_js' after 3.98 s
[04:47:05] Finished 'clean_apps' after 4.96 s
[04:47:05] Finished 'clean_dist' after 4.97 s
[04:47:05] Starting 'do_dist'...
[04:47:06] Finished 'do_dist' after 1.2 s
[04:47:06] Starting 'do_yarn'...
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Building fresh packages...
success Saved lockfile.
$ yarn _postinstall
$ platform-dependent-modules
* [platform-dependent-modules] process.cwd(): /Users/mikeller/git/blheli-configurator/dist __dirname: /Users/mikeller/git/blheli-configurator/node_modules/platform-dependent-modules/lib
* [platform-dependent-modules] npm install gulp-macdmg@0.0.1
npm WARN lifecycle The node binary used for scripts is /var/folders/nx/fgnd8mdj1vs_ddhggtx5khmc0000gn/T/yarn--1606567638949-0.5388593193601634/node but npm is using /usr/local/Cellar/node@10/10.22.1/bin/node itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.
npm notice created a lockfile as package-lock.json. You should commit this file.


> fs-xattr@0.3.1 install /Users/mikeller/git/blheli-configurator/dist/node_modules/fs-xattr
> node-gyp rebuild

  CC(target) Release/obj.target/xattr/src/async.o
  CC(target) Release/obj.target/xattr/src/error.o
  CC(target) Release/obj.target/xattr/src/sync.o
  CC(target) Release/obj.target/xattr/src/util.o
  CC(target) Release/obj.target/xattr/src/xattr.o
  SOLINK_MODULE(target) Release/xattr.node

> macos-alias@0.2.11 install /Users/mikeller/git/blheli-configurator/dist/node_modules/macos-alias
> node-gyp rebuild

  CXX(target) Release/obj.target/volume/src/volume.o
  SOLINK_MODULE(target) Release/volume.node
+ gulp-macdmg@0.0.1
added 9 packages from 7 contributors, removed 9 packages, updated 62 packages and audited 71 packages in 15.54s

1 package is looking for funding
  run `npm fund` for details

found 0 vulnerabilities


[04:47:36] Finished 'do_yarn' after 30 s
[04:47:36] Starting 'do_apps'...
Building for platform(s): osx64.
Release build.
Using v0.49.1 (normal)
Create cache folder in /Users/mikeller/git/blheli-configurator/cache/0.49.1-normal
Using cache for: osx64
Create release folder in /Users/mikeller/git/blheli-configurator/apps/blheli-configurator/osx64
[04:47:41] Finished 'do_apps' after 5.52 s
[04:47:41] Starting 'do_release'...
Building for platform(s): osx64.
Packing release.
[04:47:41] 'do_release' errored after 2.24 ms
[04:47:41] Error: Cannot find module 'gulp-appdmg'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at release_osx64 (/Users/mikeller/git/blheli-configurator/gulpfile.js:346:18)
    at do_release (/Users/mikeller/git/blheli-configurator/gulpfile.js:402:9)
    at bound (domain.js:402:14)
    at runBound (domain.js:415:12)
    at asyncRunner (/Users/mikeller/git/blheli-configurator/node_modules/async-done/index.js:55:18)
    at process._tickCallback (internal/process/next_tick.js:61:11)
[04:47:41] 'release' errored after 41 s
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
mikeller@iMac blheli-configurator % 
```

File: package.json [hunk 1]
```diff
@@ -52,8 +52,11 @@
   "config": {
     "platformDependentModules": {
       "darwin": [
-        "gulp-appdmg@1.0.3"
+        "gulp-macdmg@0.0.1"
       ]
     }
+  },
+  "dependencies": {
```
Comment (mikeller) [inline package.json:59]: This (added by `platform-dependent-modules`) cannot be committed, as this will make the build break for all platforms except for MacOS.

**Review by mikeller — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (nerdCopter) [reply package.json:59]: sorry, i know this was 2020, but `gulp-macdmg` solved a project for me. (seemingly because it may avoid `macos-alias` and `gulp-util`).

The cross-platform issue should be solvable with this:
```
  "optionalDependencies": {
    "gulp-macdmg@0.0.1"
  }
```
